sklearn/discriminant_analysis.py

```diff
@@ -864,7 +872,7 @@ def fit(self, X, y):
         if self.priors is None:
             self.priors_ = np.bincount(y) / float(n_samples)
         else:
-            self.priors_ = self.priors
+            self.priors_ = np.array(self.priors, copy=False)
```

Comment:
We should take a copy here to avoid some surprises.

```suggestion
            self.priors_ = np.array(self.priors)
```

Otherwise, I would expect something funky here:

```python
priors = np.array([0.5, 0.5])
qda = QuadraticDeterminantAnalysis(priors=priors).fit(X, y)
# we expect the following
assert qda.priors_ == qda.priors
# altering `priors` without `fit` should not change `priors_`
priors[0] = 0.2
assert qda.priors != qda.priors
```

Reply:
We could add this test to make sure that `assert not self.priors_ is self.priors`